### PR TITLE
feat: added resourceLimits to pods

### PR DESCRIPTION
**Pull request by christianhilbrands**
#### Checklist
- [x] Functionality works & meets specifications (tested on rancher)
- [x] Templates reviewed
- [ ] Added values to questions
- [x] Bumped the chart version
- [x] Updated appVersion (if needed)

### CODE REVIEW

File: charts/molgenis-jenkins/values.yaml [hunk 1]
```diff
@@ -543,6 +578,7 @@ jenkins:
               ttyEnabled: true
               resourceRequestCpu: "100m"
               resourceRequestMemory: "256Mi"
+              resourceLimitCpu: "200m"
               resourceLimitMemory: "512Mi"
       molgenis-it-jdk15: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:583]: The whole pod can be deleted `molgenis-it-jdk15`

File: charts/molgenis-jenkins/values.yaml [hunk 2]
```diff
@@ -602,6 +640,7 @@ jenkins:
               args: ""
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:640]: The whole pod can be deleted `molgenis-it-jdk15-gradle`

File: charts/molgenis-jenkins/values.yaml [hunk 3]
```diff
@@ -656,26 +700,46 @@ jenkins:
...
             - name: debian-10
               image: "debian:buster"
               ttyEnabled: true
               command: "/bin/bash"
               args: ""
+              resourceLimitCpu: "2"
+              resourceLimitMemory: "4Gi"
+              resourceRequestCpu: "1"
+              resourceRequestMemory: "1Gi"
       r-3.5.2: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:743]: Remove the whole pod `r-3.5.2`

File: charts/molgenis-jenkins/values.yaml [hunk 4]
```diff
@@ -687,11 +751,19 @@ jenkins:
...
             - name: curl
               image: "appropriate/curl:latest"
               ttyEnabled: true
               command: "cat"
               args: ""
+              resourceLimitCpu: "2"
+              resourceLimitMemory: "4Gi"
+              resourceRequestCpu: "1"
+              resourceRequestMemory: "1Gi"
       r-3.6.2: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:767]: Remove the whole pod `r-3.6.2`

File: charts/molgenis-jenkins/values.yaml [hunk 5]
```diff
@@ -703,11 +775,19 @@ jenkins:
...
             - name: curl
               image: "appropriate/curl:latest"
               ttyEnabled: true
               command: "cat"
               args: ""
+              resourceLimitCpu: "2"
+              resourceLimitMemory: "4Gi"
+              resourceRequestCpu: "1"
+              resourceRequestMemory: "1Gi"
       r-4.0.3: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:791]: Remove the whole pod `r-4.0.3`

File: charts/molgenis-jenkins/values.yaml [hunk 6]
```diff
@@ -719,11 +799,19 @@ jenkins:
...
             - name: curl
               image: "appropriate/curl:latest"
               ttyEnabled: true
               command: "cat"
               args: ""
+              resourceLimitCpu: "2"
+              resourceLimitMemory: "4Gi"
+              resourceRequestCpu: "1"
+              resourceRequestMemory: "1Gi"
       helm: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:815]: Remove the whole pod 'helm'

File: charts/molgenis-jenkins/values.yaml [hunk 7]
```diff
@@ -643,6 +683,10 @@ jenkins:
             - name: jekyll
               image: "molgenis/jekyll:jekyll-release-2019-03-06_08-18-10"
               ttyEnabled: true
+              resourceLimitCpu: "1"
+              resourceLimitMemory: "2Gi"
+              resourceRequestCpu: "1"
+              resourceRequestMemory: "1Gi"
               command: "cat"
               args: ""
       ansible: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:692]: Remove the whole pod 'ansible'

File: charts/molgenis-jenkins/values.yaml [hunk 8]
```diff
@@ -424,6 +448,8 @@ jenkins:
               command: cat
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:448]: Remove whole pod `node`

File: charts/molgenis-jenkins/values.yaml [hunk 9]
```diff
@@ -424,6 +448,8 @@ jenkins:
...
               args: ""
               ttyEnabled: true
+              resourceLimitCpu: "1"
+              resourceLimitMemory: "2Gi"
               resourceRequestCpu: "1"
               resourceRequestMemory: "1Gi"
       node-dubnium: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:455]: Remove whole pod `node-dubnium`

**Review by sidohaakma — CHANGES_REQUESTED** [review]

File: charts/molgenis-jenkins/values.yaml [hunk 10]
```diff
@@ -470,6 +500,8 @@ jenkins:
               command: cat
               args: ""
               ttyEnabled: true
+              resourceLimitCpu: "1"
+              resourceLimitMemory: "4Gi"
               resourceRequestCpu: "1"
               resourceRequestMemory: "1Gi"
       python: |
```
Comment (sidohaakma) [inline charts/molgenis-jenkins/values.yaml:507]: Remove whole pod `python`

**Review by sidohaakma — CHANGES_REQUESTED** [review]

File: charts/molgenis-jenkins/values.yaml [hunk 11]
```diff
@@ -275,11 +275,19 @@ jenkins:
...
+              resourceLimitMemory: "4Gi"
+              resourceRequestCpu: "1"
+              resourceRequestMemory: "1Gi"
             - name: rancher
               image: "rancher/cli2:v2.4.10"
               command: cat
               args: ""
               ttyEnabled: true
+              resourceLimitCpu: "2"
+              resourceLimitMemory: "4Gi"
```
[the remark this answers is not in the document]
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml:288]: here: https://www.jenkins.io/doc/pipeline/steps/kubernetes/ under podTemplate -> containers i see only resourceLimit and resourceRequest? am i looking to the wrong section?

File: charts/molgenis-jenkins/values.yaml (hunk 5 above)
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml:791]: done

File: charts/molgenis-jenkins/values.yaml (hunk 6 above)
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml:815]: done

File: charts/molgenis-jenkins/values.yaml [hunk 12]
```diff
@@ -306,6 +318,10 @@ jenkins:
...
```
[the remark this answers is not in the document]
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml:322]: done

File: charts/molgenis-jenkins/values.yaml (hunk 8 above)
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

File: charts/molgenis-jenkins/values.yaml (hunk 9 above)
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

File: charts/molgenis-jenkins/values.yaml (hunk 10 above)
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

File: charts/molgenis-jenkins/values.yaml [hunk 13]
```diff
@@ -457,6 +485,8 @@ jenkins:
               command: cat
               args: ""
               ttyEnabled: true
+              resourceLimitCpu: "1"
+              resourceLimitMemory: "4Gi"
```
[the remark this answers is not in the document]
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

File: charts/molgenis-jenkins/values.yaml [hunk 14]
```diff
@@ -275,11 +275,19 @@ jenkins:
               command: cat
               args: ""
               ttyEnabled: true
+              resourceLimitCpu: "2"
+              resourceLimitMemory: "4Gi"
```
[the remark this answers is not in the document]
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

File: charts/molgenis-jenkins/values.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

File: charts/molgenis-jenkins/values.yaml [hunk 15]
```diff
@@ -289,6 +297,10 @@ jenkins:
...
```
[the remark this answers is not in the document]
Comment (christianhilbrands) [reply charts/molgenis-jenkins/values.yaml]: done

**Comment by christianhilbrands** [comment]
the PR is closed because there were problems with the new version of jenkins with the plugins. in the job plugin was a function deprecated and stopped the whole deployment of jenkins. i have ported the necessary pod templates to https://github.com/molgenis/molgenis-ops-helm/pull/507. this need to be looked at at an later date

**Pull request closed without merging**